ichnaea/data/schema.py

```diff
@@ -0,0 +1,376 @@
+import copy
+import datetime
+import uuid
+from datetime import timedelta
+
+import mobile_codes
+from colander import (
+    Boolean,
+    DateTime,
+    Float,
+    Integer,
+    Invalid,
+    MappingSchema,
+    Range,
+    SchemaNode,
+    String,
+    iso8601
+)
+
+from ichnaea import geocalc
+from ichnaea.customjson import encode_datetime
+from ichnaea.data.constants import (
+    ALL_VALID_MCCS,
+    INVALID_WIFI_REGEX,
+    MAX_ACCURACY,
+    MAX_ALTITUDE,
+    MAX_ALTITUDE_ACCURACY,
+    MAX_HEADING,
+    MAX_LAT,
+    MAX_SPEED,
+    MIN_ALTITUDE,
+    MIN_LAT,
+    VALID_WIFI_REGEX,
+)
+from ichnaea.models import (
+    MAX_RADIO_TYPE,
+    MIN_RADIO_TYPE,
+    RADIO_TYPE,
+)
+from ichnaea import util
+
+
+# Utility functions
+
+def normalized_time(time):
+    """
+    Takes a string representation of a time value, validates and parses
+    it and returns a JSON-friendly string representation of the normalized
+    time.
+    """
```

Comment:
This docstring was bad before, but could you add a sentence to mention the whole "convert to first of month" and "magically convert dates older than two months to now" business?